internal/api/v1/resource.go

```diff
@@ -0,0 +1,187 @@
+	}
+	// Wait for the calls to finish.
+	wg.Wait()
+	// Close the channel.
+	close(uc)
+
+	// Remove existing entries from the DB.
+	err = cc.SQLClient.DeleteKubernetesResourcesByAccountName(account)
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
+		return
+	}
+
+	// Use the same task ID for all entries.
+	taskID := uuid.New().String()
+	resources := []kubernetes.Resource{}
+
+	// Create a resource entry in the DB for each unstructured object (manifest).
+	for u := range uc {
+		gvr, err := provider.Client.GVRForKind(u.GetKind())
+		if err != nil {
+			c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
+			return
+		}
+		// Map values to kubernetes.Resource.
+		nameWithoutVersion := kubernetes.NameWithoutVersion(u.GetName())
+		kr := kubernetes.Resource{
+			AccountName:  account,
+			ID:           uuid.New().String(),
+			TaskID:       taskID,
+			Timestamp:    internal.CurrentTimeUTC(),
+			APIGroup:     gvr.Group,
+			Name:         u.GetName(),
+			ArtifactName: nameWithoutVersion,
+			Namespace:    u.GetNamespace(),
+			Resource:     gvr.Resource,
+			Version:      gvr.Version,
+			Kind:         u.GetKind(),
+			SpinnakerApp: kubernetes.SpinnakerMonikerApplication(u),
+			Cluster:      kubernetes.Cluster(u.GetKind(), nameWithoutVersion),
+		}
+		// Insert row into kubernetes_resources table.
+		err = cc.SQLClient.CreateKubernetesResource(kr)
+		if err != nil {
+			c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
+			return
+		}
+
+		resources = append(resources, kr)
+	}
+
+	c.JSON(http.StatusOK, resources)
+}
+
+// DeleteKubernetesResources deletes the resources from the database
+// for the given provider (account).
+func (cc *Controller) DeleteKubernetesResources(c *gin.Context) {
+	name := c.Param("name")
+
+	_, err := cc.SQLClient.GetKubernetesProvider(name)
+	if err != nil {
+		if err == gorm.ErrRecordNotFound {
+			c.JSON(http.StatusNotFound, gin.H{"error": "provider not found"})
+			return
+		}
+
+		c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
+
+		return
+	}
+
+	err = cc.SQLClient.DeleteKubernetesResourcesByAccountName(name)
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
+		return
+	}
+
+	c.JSON(http.StatusNoContent, nil)
+}
+
+// listKinds lists a given kind and sends to a channel of unstructured.Unstructured.
+// It uses a context with a timeout of 10 seconds.
+func listKinds(wg *sync.WaitGroup, uc chan unstructured.Unstructured,
+	provider *kubernetes.Provider, kind, account string) {
+	// Finish the wait group when we're done here.
+	defer wg.Done()
+
+	// Declare server side filtering options.
+	lo := metav1.ListOptions{
+		LabelSelector: kubernetes.DefaultLabelSelector(),
+	}
+	// If namespace-scoped account, then only get resources in the namespace.
+	if provider.Namespace != nil {
+		lo.FieldSelector = "metadata.namespace=" + *provider.Namespace
+	}
```

Comment:
This is a good callout. Shouldn't we also do this for the clusters page?

https://github.com/homedepot/go-clouddriver/blob/master/internal/api/core/applications.go#L1427